climakitae/threshold_tools.py

```diff
@@ -884,3 +889,523 @@ def return_period(ams):
+                self.param.smoothing,
+                self.param.num_timesteps, 
+                width=375
+            )
+        else:
+            smooth_row = pn.Row(
+                self.param.smoothing,
+                width=375
+            )
+        return pn.Card(smooth_row, title = "Smoothing", collapsible = False)
+
+    @param.depends("duration1_length", "duration1_type", watch=False)
+    def group_row(self):
+        """A reactive row for duration2 options that updates if group is updated"""
+        self.group_length = self.duration1_length
+        self.group_type = self.duration1_type
+        return pn.Row(
+            self.param.group_length, self.param.group_type, 
+            width=375
+        )
+
+    @param.depends("group_length", "group_type", watch=False)
+    def duration2_row(self):
+        """A reactive row for duration2 options that updates if group is updated"""
+        self.duration2_length = self.group_length
+        self.duration2_type = self.group_type
+        return pn.Row(
+            self.param.duration2_length, self.param.duration2_type, 
+            width=375
+        )
+
+def _exceedance_visualize(choices, option=1):
+    """
+    Uses holoviz 'panel' library to display the parameters and view defined for exploring exceedance.
+    """
+    _left_column_width = 375
+
+    if option==1:
+        plot_card = choices.view
+    elif option==2:
+        # For show: potential option to display multiple tabs if we want to 
+        # build this out as a broader GUI app for all threshold tools
+        plot_card = pn.Tabs(
+            ("Event counts", choices.view), 
+            ("Return values", pn.Row()),
+            ("Return periods", pn.Row())
+        ) 
+    else:
+        raise ValueError("Unknown option")
+
+    options_card = pn.Card(
+        # Threshold value and direction
+        pn.Row(
+            choices.param.threshold_direction,
+            choices.param.threshold_value,
+            width = _left_column_width
+        ),
+
+        # DURATION 1
+        "I'm interested in extreme conditions that last for . . .",
+        pn.Row(
+            choices.param.duration1_length,
+            choices.param.duration1_type, width=375
+        ),
+        pn.layout.Divider(margin = (-10,0,-10,0)),
+
+        # PERIOD
+        "Show me a timeseries of the number of occurences every . . .",
+        pn.Row(
+            choices.param.period_length,
+            choices.param.period_type,
+            width = _left_column_width
+        ),
+        "Examples: for an annual timeseries, select '1-year'. For a seasonal timeseries, select '3-month'.",
+        pn.layout.Divider(margin = (-10,0,-10,0)),
+
+        # GROUP
+        "Optional aggregation: I'm interested in the number of ___ that contain at least one occurance.",
+        choices.group_row,
+
+        # DURATION 2
+        "After aggregation, I'm interested in occurances that last for . . .",
+        choices.duration2_row,
```

Comment:
It's not clear to me by the wording for this selection if this is being performed or not. It says "optional", but a selection is shown. The default shows 2hr, and 2hr, but is labelled as an optional selection, making me unsure whether this aggregation is occurring even if I haven't modified the values?